packages/codegen/src/generateEntityCodegenFile.ts

```diff
@@ -238,6 +244,47 @@ export function generateEntityCodegenFile(config: Config, meta: EntityDbMetadata
   `;
 }
 
+function fieldHasDefaultValue(
+  config: Config,
+  meta: EntityDbMetadata,
+  field: EntityDbMetadata["primitives"] extends Array<infer U> ? U : never,
```

Comment:
Huh, what's this extends + infer doing? Are we trying to constrain that the runtime `field` value of comes the `meta.primitives` column? I'm not sure we have the type information to enforce that.

Reply:
primitives is defined as `PrimitiveField[]`  but we don't export `PrimitiveField` from in `EntityDbMetadata.ts`, so this allows us to use that type without needing to export it.  I could just add the export if you'd rather i do that.

Reply:
Ah cool. Does just `EntityDbMetadata["primitives"]` work? I was mostly curious about the `Array<infer U>` seeming unnecessary/or else I'm missing something.

Reply:
No, because EntityDBMetaData is an array.  So we need to extract out the generic argument of the array with infer.  This was the only syntax i could get to work, although i've just tried `EntityDbMetadata["primitives"] extends (infer U)[] ? U : never` and it also works.  Do you have a preference?  Just exporting the type is also still an option.

Reply:
I think `EntityDBMetadata["primitives][]` or `EntityDBMetadata["primitives"][0]` is supposed to work? I forget the exact incantation.

Reply:
Neat, `EntityDBMetadata["primitives"][0]` does actually work as well.  I'm not sure which option is the most clear/least magical, though:

`EntityDBMetadata["primitives"][0]`
`EntityDBMetadata["primitives"] extends (infer U)[] ? U : never`
`EntityDBMetadata["primitives"] extends Array<infer U> ? U : never`

or, in 

`export type PrimitiveField  = ...` in `EntityDbMetadata.ts`

Reply:
`EntityDBMetadata["primitives"][0]` is pretty okay-ish/idiomatic in TS code, once you learn/see it b/c its somewhat unintuitive, so I'm +1 with that, but also +1 to the exported type.

Reply:
Gonna go with just exporting all those types.  I'm not sure there's any benefit to them being private from the rest of joist.